### PR TITLE
Nil URL Crash

### PR DESCRIPTION
**Pull request by shortstuffsushi**
@aaustin @derrickstaten

### CODE REVIEW

**Comment by aaustin** [comment]
nice :+1:

File: Branch-SDK/Branch-SDK/Branch.m [hunk 1]
```diff
@@ -1694,9 +1694,8 @@ - (void)serverCallback:(BNCServerResponse *)response {
             }
             
             // cache the link
-            BNCLinkData *linkData = response.linkData;
-            if (linkData) {
-                [self.linkCache setObject:url forKey:linkData];
+            if (url) {
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: how about if (url && linkData)

Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/Branch.m]: Link Data cannot be nil.

File: Branch-SDK/Branch-SDK/Branch.m [hunk 2]
```diff
@@ -1038,11 +1038,11 @@ - (NSString *)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andTyp
             shortURL = [serverResponse.data objectForKey:URL];
             
             // cache the link
-            BNCLinkData *linkData = serverResponse.linkData;
-            if (linkData) {
-                [self.linkCache setObject:shortURL forKey:linkData];
+            if (shortURL) {
```
Comment (qinweigong) [inline Branch-SDK/Branch-SDK/Branch.m]: same here, check both?

**Pull request merged**